packages/transport/src/transports/usb.ts

```diff
@@ -0,0 +1,332 @@
+    }
+
+    public acquire({ input }: { input: AcquireInput }) {
+        return new AbortablePromise<
+            Success<string>,
+            ErrorGeneric<
+                | typeof INTERFACE_ERRORS.UNABLE_TO_CLAIM_INTERFACE
+                | typeof INTERFACE_ERRORS.DEVICE_DISCONNECTED_WEBUSB
+                | typeof SESSION_ERRORS.WRONG_PREVIOUS_SESSION
+            >
+        >(async (resolve, reject) => {
+            // listenPromise is resolved on next listen
+            this.listenPromise = createDeferred();
+
+            const acquireIntentResponse = await this.sessionsClient.acquireIntent(input);
+            console.log('acquireIntentResponse', acquireIntentResponse);
+            if (!acquireIntentResponse.success) {
+                // connect expects us to throw here
+                return reject(this.error({ error: acquireIntentResponse.error }));
+            }
+
+            this.acquiringPath = input.path;
+
+            const reset = !!input.previous;
+
+            const { path } = input;
+
+            const openDeviceResult = await this.transportInterface.openDevice(path, reset);
+
+            this.log('openDeviceResult', openDeviceResult);
+
+            if (!openDeviceResult.success) {
+                return reject(openDeviceResult);
+            }
+
+            const acquireDoneResult = await this.sessionsClient.acquireDone({ path });
+            this.log('acquireDoneResult', acquireDoneResult);
+
+            const expectedSessionId = acquireDoneResult.payload.session;
+            this.acquiringSession = expectedSessionId;
+
+            if (!this.listening) {
+                return resolve(this.success(expectedSessionId));
+            }
+
+            return this.listenPromise.promise
+                .then(sessionId => {
+                    resolve(this.success(sessionId));
+                    delete this.listenPromise;
+                })
+                .catch(err => {
+                    reject(this.error(err));
+                    delete this.listenPromise;
+                });
+        }, this.abortController);
+    }
+
+    call({
+        session,
+        name,
+        data,
+    }: {
+        session: string;
+        name: string;
+        data: Record<string, unknown>;
+    }) {
+        return new AbortablePromise<Success<MessageFromTrezor>, Error>(async (resolve, reject) => {
+            const getPathBySessionResponse = await this.sessionsClient.getPathBySession({
+                session,
+            });
+            if (!getPathBySessionResponse.success) {
+                // session not found means that device was disconnected
+                if (getPathBySessionResponse.error === 'session not found') {
+                    return reject(this.error({ error: 'device disconnected during action' }));
+                } else {
+                    // should never happen
+                    return reject(this.error({ error: 'unexpected error' }));
+                }
```

Comment:
Since there is a return in the previous if, this else should be unnecessary:

```suggestion
                }
               // should never happen
               return reject(this.error({ error: 'unexpected error' }));
```